service/history/mutableStateBuilder.go

```diff
@@ -91,10 +106,13 @@ type (
 		hBuilder         *historyBuilder
 
 		// in memory only attributes
+		// indicate the current version
+		currentVersion int64
 		// indicates whether there are buffered events in persistence
 		hasBufferedEventsInDB bool
-		// indicates whether the workflow is pointed by current record, guaranteed
-		isWorkflowBeingCurrentGuaranteed bool
+		// indicates the workflow state in DB, can be used to calculate
+		// whether this workflow is pointed by current workflow record
+		stateInDB int
```

Comment:
Do we really want to emphasis "InDB"?

Reply:
yes, nextEventIDInDB is used for compare and swap

stateInDB is used like a bloom filter, i.e. stateInDB being created / running meaning that this workflow must be the current workflow (assuming there is no rebuild of mutable state).

stateInDB being completed does not guarantee this workflow being the current workflow
stateInDB being zombie guarantees this workflow not being the current workflow